testsuite/testsuite-microprofile-jwt/src/test/java/org/wildfly/swarm/microprofile/jwtauth/RolesAllowedEmptyPropsTest.java

```diff
@@ -0,0 +1,115 @@
+import java.util.Base64;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.UrlAsset;
+import org.jboss.shrinkwrap.impl.base.io.IOUtil;
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.swarm.jaxrs.JAXRSArchive;
+
+@RunWith(Arquillian.class)
+public class RolesAllowedEmptyPropsTest {
+
+    @Deployment
+    public static Archive<?> createDeployment() throws Exception {
+        JAXRSArchive deployment = ShrinkWrap.create(JAXRSArchive.class);
+        deployment.addResource(RolesEndpointClassLevel.class);
+        deployment.addResource(TestApplication.class);
+        deployment.addAllDependencies();
+        deployment.addAsResource("emptyRoles.properties");
+        deployment.addAsResource("project-defaults.yml");
+        URL url = RolesAllowedEmptyPropsTest.class.getResource("/publicKey.pem");
+        deployment.addAsManifestResource(new UrlAsset(url), "/MP-JWT-SIGNER");
+        return deployment;
+    }
+
+    @RunAsClient
+    @Test
+    public void testRolesAllowed() throws Exception {
+        String uri = "http://localhost:8080/mpjwt/rolesClass";
+        WebTarget target = ClientBuilder.newClient().target(uri).queryParam("input", "hello");
+        String response = target.request(MediaType.TEXT_PLAIN)
+            .header(HttpHeaders.AUTHORIZATION, "Bearer " + createToken("Echoer")).get(String.class);
+        Assert.assertEquals(response, "hello, user=jdoe@example.com");
+    }
+    
+    @RunAsClient
+    @Test
+    public void testRolesNotAllowed() throws Exception {
+        String uri = "http://localhost:8080/mpjwt/rolesClass";
+        WebTarget target = ClientBuilder.newClient().target(uri).queryParam("input", "hello");
+        Response resp = target.request(MediaType.TEXT_PLAIN)
+            .header(HttpHeaders.AUTHORIZATION, "Bearer " + createToken("Echoer2")).get();
+        Assert.assertEquals(Response.Status.FORBIDDEN, resp.getStatusInfo());
+    }
+    
+    private static String createToken(String groupName) throws Exception {
+        String value = "{"
+                + "\"iss\": \"https://server.example.com\","
+                + "\"sub\": \"24400320\","
+                + "\"exp\": " + System.currentTimeMillis() + 36000 + ","
+                + "\"upn\": \"jdoe@example.com\","
+                + "\"groups\": ["
+                + " \"" + groupName + "\"" 
+                + " ]"
+                + "}";
```

Comment:
Doesn't jose4j have a builder to construct the token in a type-safe fashion?

Reply:
Let me try...